src/openrct2/ride/Vehicle.cpp

```diff
@@ -9296,39 +9309,36 @@ int32_t Vehicle::UpdateTrackMotionMiniGolf(int32_t* outStation)
         poweredAcceleration *= powered_acceleration << 1;
         poweredAcceleration = poweredAcceleration / quarterForce;
 
-        if (!(vehicleEntry->flags & VEHICLE_ENTRY_FLAG_WATER_RIDE))
+        if (vehicleEntry->flags & VEHICLE_ENTRY_FLAG_WATER_RIDE)
         {
-            goto loc_6DD054;
-        }
-
-        if (poweredAcceleration < 0)
-        {
-            poweredAcceleration >>= 4;
-        }
-
-        if (vehicleEntry->flags & VEHICLE_ENTRY_FLAG_SPINNING)
-        {
-            spin_speed = std::clamp(spin_speed, VEHICLE_MIN_SPIN_SPEED_WATER_RIDE, VEHICLE_MAX_SPIN_SPEED_WATER_RIDE);
-        }
+            if (poweredAcceleration < 0)
+            {
+                poweredAcceleration >>= 4;
+            }
 
-        if (vehicle_sprite_type != 0)
-        {
-            poweredAcceleration = std::max(0, poweredAcceleration);
             if (vehicleEntry->flags & VEHICLE_ENTRY_FLAG_SPINNING)
             {
-                if (vehicle_sprite_type == 2)
+                spin_speed = std::clamp(spin_speed, VEHICLE_MIN_SPIN_SPEED_WATER_RIDE, VEHICLE_MAX_SPIN_SPEED_WATER_RIDE);
+            }
+
+            if (vehicle_sprite_type != 0)
+            {
+                poweredAcceleration = std::max(0, poweredAcceleration);
+                if (vehicleEntry->flags & VEHICLE_ENTRY_FLAG_SPINNING)
                 {
-                    spin_speed = 0;
+                    if (vehicle_sprite_type == 2)
+                    {
+                        spin_speed = 0;
+                    }
                 }
+                newAcceleration += poweredAcceleration;
+                goto loc_6DD069;
```

Comment:
I get why you did this, but I'd prefer if you didn't introduce a goto.

Reply:
Neither would I but it will be easier to refactor the two together now.

Reply:
Do you plan to do that in this PR or in another?

Reply:
In another refactor